olive/passes/onnx/extract_adapters.py

```diff
@@ -0,0 +1,336 @@
+                            packings[f"{base_name}.quant{suffix}.packed"] = [
+                                name + suffix for name in matching_quant_modules if name + suffix in weights
+                            ]
+
+            # pack the weights, create inputs and split nodes
+            packed_weights = {}
+            for weight_name, to_pack in packings.items():
+                if not to_pack:
+                    continue
+                packed_weights[weight_name] = np.concatenate([np.atleast_1d(weights[name]) for name in to_pack], axis=0)
+
+                # input proto
+                input_proto = onnx.helper.make_tensor_value_info(
+                    name=weight_name,
+                    elem_type=onnx.helper.np_dtype_to_tensor_dtype(packed_weights[weight_name].dtype),
+                    shape=packed_weights[weight_name].shape,
+                )
+                # TODO(jambayk): check if graph_idx can be 0 even if they might be used in subgraphs
+                dag.add_input(input_proto, 0)
+
+                # split nodes
+                split_node_proto = onnx.helper.make_node(
+                    "Split",
+                    inputs=[weight_name],
+                    outputs=to_pack,
+                    name=f"{weight_name}.split",
+                    axis=0,
+                )
+                dag.add_node(split_node_proto, 0, overwrite_initializers=True)
+
+            # remove the original weights
+            weights = packed_weights
+
+        # update the model with the changes
+        dag.update()
+
+        # save the weights
+        # TODO(jambayk): Consider other methods for saving the weights
+        # safetensors is an option but it is not available on ARM64 Windows
+        weights_path = Path(output_model_path).parent / "adapter_weights.npz"
+        np.savez(weights_path, **weights)
+
+        # save the model
+        output_model = model_proto_to_olive_model(
+            dag.model,
+            output_model_path,
+            external_data_config={"save_as_external_data": True, "all_tensors_to_one_file": True},
+            external_initializers_name=weights_path.name if not config["make_inputs"] else None,
+            constant_inputs_name=weights_path.name if config["make_inputs"] else None,
+        )
+        output_model.model_attributes = deepcopy(model.model_attributes)
+        # save information about the weights in the model attributes
+        weights_info = {name: [list(value.shape), str(value.dtype)] for name, value in weights.items()}
+        if not config["make_inputs"]:
+            output_model.model_attributes["external_initializers"] = weights_info
+        else:
+            output_model.model_attributes["constant_inputs"] = weights_info
+            if config["pack_inputs"]:
+                output_model.model_attributes["packed_inputs"] = packings
+        return output_model
+
+    @staticmethod
+    def remove_identity_nodes(dag: OnnxDAG):
```

Comment:
should it be method of OnnxDag?

Reply:
Yes, I can make this a method of OnnxDAG since it only deals with the graph and can be reused for other passes too.

Reply:
done.